src/main/java/seedu/duke/command/AddToSemCommand.java

```diff
@@ -79,17 +79,22 @@ private void checkAvailableModulesList(SelectedModule selectedModule) {
     }
 
     /**
-     *Allows users to add to data later(or not) and then it if users add a module to data,
-     *it will automatically update the information of the selected module.
-     * @param semesterList user's current semester list
+     * Checks if the selected module being added is exist in any of the semester list
+     * If selected module exist inside semester list, it will check if the module inside the list has failed.
+     * It will return false to allow the method to be added into semester list.
+     * @param semesterList user's semester list
      * @return boolean value of true if the module is in the user's semester list, and false otherwise
      */
     private boolean checkModuleExist(SemesterList semesterList) {
         for (SemModulesList sem: semesterList) {
-            if (sem.isInList(selectedModule.getName()) || sem.isInList(selectedModule.getId())) {
-                return true;
+            for (SelectedModule module: sem) {
+                boolean hasSameModuleId = module.getId().equals(selectedModule.getId()) && module.isIdValid();
+                boolean hasSameModuleName = module.getName().equals(selectedModule.getName()) && module.isNameValid();
+                if (hasSameModuleId || hasSameModuleName) {
+                    return true;
+                    }
+                }
```

Comment:
These two parts(deleted and new added) seem to have the same function or meaning haha.